source/common/router/router.cc

```diff
@@ -333,7 +333,11 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
 
   const Http::HeaderEntry* request_alt_name = headers.EnvoyUpstreamAltStatName();
   if (request_alt_name) {
-    alt_stat_prefix_ = std::string(request_alt_name->value().getStringView()) + ".";
+    // NOTE: converting this header value into a StatName requires taking a
+    // global symbol-table lock. Is this scenario common enough to raise
+    // concern?  Or is this effectively a test hook?
```

Comment:
This is not a test feature. People do use this in production, but I'm not sure how often it's actually used. I think it's probably fine but I would maybe put in a TODO about thinking about improving this in the future and maybe put a warning in the docs on the documentation for this header? WDYT?

Reply:
Got it. Found the doc:

```
./docs/root/configuration/http_filters/router_filter.rst
./docs/root/configuration/http_conn_man/header_sanitizing.rst
```

Since we are still using FakeSymbolTable this will not take a lock yet, so this PR doesn't introduce a lock in the hot-path.

Maybe I should add a tls-cache for cases where we really need to compose new StatName from string in the hotpath. I'm thinking there may be a few cases like this, which would justify adding this capability, rather than having to explain lock overhead details in user doc. Since we are still using Fakes I'd prefer to do that in a separate PR.

 WDYT?

Reply:
Yeah I think let's just put in a TODO for now and not add any docs. I don't think very many people are using this feature and I doubt they would notice the lock overhead even if we added it.

With that said, although I'm pretty certain this change is going to make this change faster, would it be worth it to add a microbenchmark for this code and then we can see the perf change?

Reply:
Actually I have a microbenchmark, and I've been running it on this in various forms: test/common/http/codes_speed_test.cc. This makes things faster with real symbol tables, but it's a wash with fake symbol tables (since that really just degenerates to the same string concatenations in the hot-path).

Reply:
Added #7003 and made this note a TODO(#7003).

Reply:
I have a draft PR which provides a mechanism to fix this, but I don't know if it's worth the complexity for this feature. https://github.com/envoyproxy/envoy/pull/7008

My feeling is that I'll leave that in draft mode until I see if there's more need for this beyond the header-controlled stat-prefix.